src/test/java/GameFieldTest.java

```diff
@@ -8,40 +8,36 @@
 public class GameFieldTest {
 
     @Test
-    public void givenGameObjectOutOfBoundary_whenAddObject_thenCorrect(){
+    public void should_notAddObject_when_objectOutOfBoundary(){
         GameField gameField = new GameField(new Point(10, 10));
 
         gameField.addObject(new GameObject(new Point(9, 9), new Point(2, 2)));
 
         Collection<GameObject> gameObjects = gameField.getGameObjects();
```

Comment:
boolean gameField.getObjectCount()

Reply:
is it okay to create methods in class only for using in test methods?

Reply:
I have not noticed that: gameField.getGameObjects() is used only for testing. remove it.

Reply:
gameField.addObject() method should throw two different exceptions when inserted object is out of boundary or intersects another object

Reply:
Code should be written to solve current task not future one